internal/pkg/aws/cloudformation/cloudformation.go

```diff
@@ -200,13 +200,32 @@ func (c *CloudFormation) Exists(name string) (bool, error) {
 	return true, nil
 }
 
-// Metadata returns the Metadata property of the CloudFormation stack's template.
-func (c *CloudFormation) Metadata(name string) (string, error) {
-	out, err := c.GetTemplateSummary(&cloudformation.GetTemplateSummaryInput{
-		StackName: aws.String(name),
-	})
+// MetadataOpts sets up optional parameters for Metadata function.
+type MetadataOpts func() *cloudformation.GetTemplateSummaryInput
+
+// WithStackName sets up the stack name for cloudformation.GetTemplateSummaryInput.
+func WithStackName(name string) MetadataOpts {
```

Comment:
>  from the fn name I would think this it could be applied to all cloudformation methods but it's only limited to Metadata.

Hmm isn't the output type clear enough to tell it is just for Metadata? Compared with
```
func (c *CloudFormation) Metadata(stackName string, opts MetadataOpts)
```
I think I'd rather to just
```
func (c *CloudFormation) Metadata(stackName string, stackSetName string)
```
But I still think it is worth to use the `MetadataOpts` so that we can enforce the correct usage of this function.

Reply:
My only issue is reserving the names `WithStackName` and `WithStackSetName` to only the metadata method. These names make you think they could apply to the `Describe` method as well but it's not true for example.  

I would rather  create separate methods:
```
Metadata(stackName string)
StackSetMetadata(stackSetName string)
```
Or rename `WithStackName` and `WithStackSetName` to `MetadataWithStackName` and `MetadataWithStackSetName` respectively.

Reply:
Ok makes sense to me. I'll make the change.